src/internal/compilerSupport/v2.ts

```diff
@@ -320,22 +320,22 @@ export function callFunction(rawFnArgs: Array<any>, def: FunctionCallDef, script
             };
         });
 
-        // registering handler for particle timeout
-        peer.internals.regHandler.timeout(particle.id, () => {
-            reject(`Request timed out for ${def.functionName}`);
+        peer.internals.initiateParticle(particle, (stage) => {
+            if (def.returnType.tag === 'void' && (stage.stage === 'sent' || stage.stage === 'localWorkDone')) {
+                resolve(undefined);
+            }
+
+            if (stage.stage === 'expired') {
+                reject(`Request timed out for ${def.functionName}`);
```

Comment:
It would be nice to show the timeout. Like `Request timed out after ${TTL} for ${def.functionName}`

Reply:
We don't know actual TTL at this point. If it not specified explicitly `particle.ttl` is undefined here. The actual value is taken either from the peer configuration or from the default value of 7000

Reply:
I would suggest to add a separate issue and implement it separately

Reply:
Can't you pass it as `stage` field?